course-material/advanced-querying-techniques/subqueries.md

```diff
@@ -1,14 +1,316 @@
+    JOIN s1.district d ON a.district_id = d.district_id
+    WHERE l.status = 'A'
+    GROUP BY a.district_id, d.district_name
+    ORDER BY count DESC
+    LIMIT 1) as A_max
+UNION ALL
+SELECT district_id, district_name, 'B' as status, count
+FROM
+    (SELECT a.district_id, d.district_name, COUNT(*) as count
+    FROM s1.account a
+    JOIN s1.loan l ON a.account_id = l.account_id
+    JOIN s1.district d ON a.district_id = d.district_id
+    WHERE l.status = 'B'
+    GROUP BY a.district_id, d.district_name
+    ORDER BY count DESC
+    LIMIT 1) as B_max
+UNION ALL
+SELECT district_id, district_name, 'C' as status, count
+FROM
+    (SELECT a.district_id, d.district_name, COUNT(*) as count
+    FROM s1.account a
+    JOIN s1.loan l ON a.account_id = l.account_id
+    JOIN s1.district d ON a.district_id = d.district_id
+    WHERE l.status = 'C'
+    GROUP BY a.district_id, d.district_name
+    ORDER BY count DESC
+    LIMIT 1) as C_max
+UNION ALL
+SELECT district_id, district_name, 'D' as status, count
+FROM
+    (SELECT a.district_id, d.district_name, COUNT(*) as count
+    FROM s1.account a
+    JOIN s1.loan l ON a.account_id = l.account_id
+    JOIN s1.district d ON a.district_id = d.district_id
+    WHERE l.status = 'D'
+    GROUP BY a.district_id, d.district_name
+    ORDER BY count DESC
+    LIMIT 1) as D_max
+```
+</details>
+<!-- #endregion -->
+
+### Question 3 (Bonus):
+Output the `COUNT()` of of each unique 'status' variable under `s1.loan` that are greater than the average of 'A' type loans. Have the outputs be only five columns for each 'status' type with a single value each alongside with the total number of loans. You must use one or more subqueries.
+
+<b>Hint</b> `CASE WHEN` is a clause that acts as a conditional statement when performing other SQL actions. Try to see how you can incorporate `CASE WHEN` with `COUNT()` to answer this question.
+
+<!-- #region -->
+<details>
+<summary>Answers</summary>
+
+For each `SELECT` argument, we are finding the count of each status using `CASE WHEN` to have a "1" count when aggregating the 'status' values. This allows our query to have the correct counts for each value given that the loan amount is less than the average 'A' loan amount. 
+
+```{code-cell} ipython3
+%%sql 
+SELECT COUNT(CASE WHEN status = 'A' THEN 1 END) AS A,
+       COUNT(CASE WHEN status = 'B' THEN 1 END) AS B,
+       COUNT(CASE WHEN status = 'C' THEN 1 END) AS C,
+       COUNT(CASE WHEN status = 'D' THEN 1 END) AS D,
+       COUNT(*) AS Total
+FROM s1.loan
+WHERE amount > (
+        SELECT AVG(amount)
+        FROM s1.loan
+        WHERE status = 'A'
+    )
+
+```
+</details>
+<!-- #endregion -->
+
+
+## Wrapping Up
+
+In this section, we introduced subqueries and how they can be implemented with common clauses. This powerful, easy to understand technique spans from being a potential replacement for joins to allowing users to use results from another query as an inner query for an outer query.
```

Comment:
The second sentence seems a bit tedious to grasp maybe because of the number of prepositions. I like bullet points in this section for brevity, if not those then rewording the second sentence would be sweet

Reply:
The second sentence has been revised. Please take a look and tell me what you think.